### DISCUSSION TITLE
Viper install is failing with "cannot find package" error

### DISCUSSION THREAD

**Discussion by uberclerk** [spf13/viper#1080] [Q&A] [answered]
cannot find package "github.com/hashicorp/hcl/tree/hcl1" in any of:
	/usr/local/Cellar/go/1.15.7_1/libexec/src/github.com/hashicorp/hcl/tree/hcl1 (from $GOROOT)
	/Users/user/go/src/github.com/hashicorp/hcl/tree/hcl1 (from $GOPATH)

MacOS 11.2.1
go 1.15.7

Looks like HCL printer may be a v1 item, which is failing to fetch with the latest Viper.  Unsure if there's a PR for HCL v2 in the works, or if HCL can be left out of use (currently using yaml).

If there's an easy fix, certainly let me know.  I'm running into this wall as I get off the ground with very little code.  Manually installing HCL does not seem to fix it.  Viper is not pulling down its dependency either.

**Comment by sagikazarmark** [comment, marked as answer]
The error suggests that you are not using [Go Modules](https://github.com/golang/go/wiki/Modules), but it's a requirement for Viper.

Read more in the [troubleshooting guide](https://github.com/spf13/viper/blob/master/TROUBLESHOOTING.md#cannot-find-package).

**Reply by uberclerk** [reply]
Thank you!

**Comment by Sergey-59** [comment]
I have the same problem when I build a docker-compose container...
cannot find package "github.com/hashicorp/hcl/hcl/printer" in any of:
        /usr/local/go/src/github.com/hashicorp/hcl/hcl/printer (from $GOROOT)
        /go/src/github.com/hashicorp/hcl/hcl/printer (from $GOPATH)

**Reply by sagikazarmark** [reply]
Please see my answer.
